examples/golang/add-object.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+)
+
+const (
+	id  = "AWS_ACCESS_KEY_ID"
+	key = "AWS_SECRET_ACCESS_KEY"
+)
+
+func main() {
+	endPoint := flag.String("endpoint", "", "region of request")
+	accessId := flag.String("accessId", "", "Access Id")
+	accessKey := flag.String("accessKey", "", "Access Key")
+	ObjectName := flag.String("objectPath", "", "Object Path")
+	bucketName := flag.String("bucketName", "", "The name of the s3 bucket")
+	flag.Parse()
+	if flag.NFlag() < 4 {
+		log.Fatalf("Usage error: EndPoint, Access ID, Access Key, and Object Path not set.")
+	}
+	// I want it to be a one way flow for the app user thats why I used flags for access key and Id instead of
+	// a direct environment variable.
+	setEnv(id, *accessId)
```

Comment:
do you have to set the credentials as env variables?
is it possible to pass them as arguments to the session creation function?
(note that setting them as env variables may create issues when the binary is executed in parallel with different credentials)

Reply:
understood. Didn't think about this edge case.